### ISSUE TITLE
nxtrim:  command not found

### ISSUE THREAD

**Issue by robalba1** [sequencing/NxTrim#26]
Hello,

I am probably missing something obvious here.  My NxTrim-0.4.1 install appears to go fine, and I have re-installed it in different locations as a sanity check, but I cannot seem to run what I think is a nxtrim standard run command.  For example, if I run:

nxtrim -1  /media/rob/4TB_HD/rawTM_Pcrispum_3.5kb.ACAGTG.L005_1P.fastq.gz -2  /media/rob/4TB_HD/rawTM_Pcrispum_3.5kb.ACAGTG.L005_2P.fastq.gz -O /media/rob/4TB_HD/nxtrim --norc --separate --preservemp-mp

Then the program immediately throws an error:  nxtrim:  command not found.  I am in the NxTrim-0.4.1 directory when I run the command and I can see the nxtrim program (112.5 kB Program file) and the mergeReads program (59.5 kB Program file) in the NxTrim-0.4.1 directory too.

I also get the same error when I try something simple with the example data:

nxtrim -1 /example/MP_R1.fastq.gz -2 /example/MP_R2.fastq.gz -O /home/rob/.linuxbrew/Cellar/NxTrim-0.4.1/example

Again, the program immediately throws an error:  nxtrim:  command not found.  

I have reloaded all my BOOST libraries, so do not think it is a problem with BOOST (as mentioned in a previous "issue."

Any suggestions?

Rob

**Comment by odoland** [comment]
Try including the path to the Nxtrim. My NxTrim was downloaded and compiled in this directory:
$pwd
/Users/odoland/software/Nxtrim/

I think yours is in /home/rob/.linuxbrew/Cellar/NxTrim-0.4.1/

Then to run it, I would try =/home/rob/.linuxbrew/Cellar/NxTrim-0.4.1/Nxtrim -1 /example/MP_R1.fastq.gz -2 /example/MP_R2.fastq.gz -O /home/rob/.linuxbrew/Cellar/NxTrim-0.4.1/example

There's a way to run it with just typing Nxtrim. I forget how, but I think it may be using an export PATH command or $cp /home/rob/.linuxbrew/Cellar/NxTrim-0.4.1/Nxtrim /usr/local/bin.

**Comment by robalba1** [comment]
Thanks odoland.  That seemed to help move things forward, but now it throws
another error:

Problem reading
/media/rob/4TB_HD/rawTM_Pcrispum_5-7kb-GCCAAT-L005_1P.fastq.gz.

I have tried to run it with the .fastq file set as well, and it throws the
same error.

Are the dashes ("-") or the underscores ("_") in my file name the issue?

Suggestions?

Rob

**Comment by odoland** [comment]
I don't think the "-" or "_" are the issues. Have you tried NxTrim on the example data set?

I made two bogus files: blahblah.R1.fastq.gz and blahblah.R2.fastq.gz
The contents of each file are just:
"Hello World!"

and then Nxtrim -1 blahblah.R1.fastq.gz -2 blahblah.R2.fastq.gz  was still able to work (although nothing was trimmed).

So I think it could be a permission error with your file? Maybe not allowed access?

I tried the command:  (which changes permissions)
$chmod 000 on blahblah.R1.fastq.gz
and I was able to replicate your problem.

"Problem reading: blahblah.R1.fastq.gz"

So maybe try to change permissions:
https://en.wikipedia.org/wiki/Chmod

Try this command: The 777 allows read,write and execute
$chmod 777 /media/rob/4TB_HD/rawTM_Pcrispum_5-7kb-GCCAAT-L005_1P.fastq.gz.

Hope that works!
